galaxy_importer/loaders/legacy_role.py

```diff
@@ -46,13 +46,18 @@ def load(self):
         if self.cfg.run_ansible_lint:
             self._lint_role()
 
-        return schema.LegacyImportResult(
-            namespace=self.namespace,
-            name=self.name,
-            metadata=self.metadata,
-            readme_file=self.readme.name,
-            readme_html=self.readme_html,
-        )
+        try:
+            result = schema.LegacyImportResult(
+                namespace=self.namespace,
+                name=self.name,
+                metadata=self.metadata,
+                readme_file=self.readme.name,
+                readme_html=self.readme_html,
+            )
+        except ValueError as e:
+            raise exc.ImporterError(str(e)) from e
```

Comment:
The legacyrole schema is following the collection schema pattern for calling exceptions, which is good, but i do think there is a minor improvement we can make… to make metadata schema errors be raised by the actual module doing schema validation:

- Rename the `LegacyMetadataParserError` to `LegacyRoleSchemaError`
- In schema.py under classes `LegacyImportResult`, `LegacyMetadata`, `LegacyGalaxyInfo`, for every `raise ValueError` change to `raise LegacyRoleSchemaError`
- This allows us to remove the 2 try/except/raise `LegacyMetadataParserError`/`ImporterError` in legacy_role.py

@EamonTracey what do you think about that?

Reply:
I think that's reasonable ... why transform every `ValueError` into an `ImporterError` with `try/except/raise from` `instead of raising directly